wcfsetup/install/files/lib/system/template/TemplateScriptingCompiler.class.php

```diff
@@ -819,17 +819,13 @@ protected function compileForeachTag($foreachTag) {
 		$foreachData['itemVar'] = $itemVar;
 		
 		$phpCode .= "\$this->foreachVars['{$hash}'] = [];\n";
-		$phpCode .= "if (isset({$itemVar})) {\n";
-		$phpCode .= "\$this->foreachVars['{$hash}']['item'] = {$itemVar};\n";
-		$phpCode .= "}\n";
+		$phpCode .= "\$this->foreachVars['{$hash}']['item'] = {$itemVar} ?? null;\n";
```

Comment:
This will introduce a behavioral change, because `item` will now always be filled. Thus this will "leak" variables containing `null` after the loop exits.

I suggest to simply replace the `isset()` with `array_key_exists` in all the modified locations.

Reply:
`array_key_exists` doesn't work here, because `$args['item']` can be a variable (e.g. `{foreach item=$foo}`).

Reply:
> Our solution was to use `array_key_exists()`, but since this requires different parameters, we ran into issues when the key or item was not a plain string, but a variable itself, e.g. `item=foo` vs `item=$foo`.

I have also explained that the "leaking" as `null` is an acceptable trade-off.

Reply:
Ah, sorry. I did not read the explanation once more after looking at it this morning and forgot about those paragraphs :sweat_smile: